### DISCUSSION TITLE
Smartcrop: Can I get a region (left, top), but not immediately apply the changes?

### DISCUSSION THREAD

**Discussion by Mereng** [libvips/libvips#2501] [Q&A] [answered]
Hello! 

I would like to be able not to apply cropping right away, but to save the region for later use. Is there such an api now? I didn't find it in the documentation, maybe it's not so obvious to me. 

Thank you!

**Comment by jcupitt** [comment, marked as answer]
Hi @Mereng,

Sure, the width and height of the output image is the size of the windows, and xoffset/yoffset is the position. libvips is lazy, so if you don't use the crop, it won't happen.

```
$ python3
Python 3.9.7 (default, Sep 10 2021, 14:59:43) 
[GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyvips
>>> x = pyvips.Image.new_from_file("k2.jpg")
>>> y = x.smartcrop(100, 100)
>>> [y.xoffset, y.yoffset, y.width, y.height]
[-1037, -846, 100, 100]
```

**Reply by Mereng** [reply]
It's work. Thank you!
